src/control/controlobjectscript.h

```diff
@@ -22,8 +22,8 @@ class ControlObjectScript : public ControlProxy {
             return m_scriptConnections.first(); };
     void disconnectAllConnectionsToFunction(const QJSValue& function);
 
-    // Called from update();
-    void emitValueChanged() override {
+    /// Called from `ControlEngineScriptLegacy::trigger`
+    void emitValueChanged() {
```

Comment:
Rename to `emitTrigger()` while we are at it?

Reply:
Or just `trigger()`?

Reply:
Should this become a private method and ControlEngineScriptLegacy as friend class then?

Reply:
> Should this become a private method and ControlEngineScriptLegacy a friend class then?

Hmm, I don't see a big problem with this method being part of the public interface. It may as well be used by the future controller engine implementation (and this class is only used in controller engines anyway).

Reply:
Then we should not mention by whom it is used in a comment, because public methods are supposed to be used by anyone.